static/js/recent_topics_ui.js

```diff
@@ -498,26 +574,43 @@ export function update_filters_view() {
     topics_widget.hard_redraw();
 }
 
-function stream_sort(a, b) {
-    const a_stream = message_store.get(a.last_msg_id).stream;
-    const b_stream = message_store.get(b.last_msg_id).stream;
-    if (a_stream > b_stream) {
+function sort_comparator(a, b) {
+    // compares strings in lowercase and returns -1, 0, 1
+    if (a.toLowerCase() > b.toLowerCase()) {
         return 1;
-    } else if (a_stream === b_stream) {
+    } else if (a === b) {
         return 0;
     }
     return -1;
 }
 
+function stream_sort(a, b) {
+    if (a.type === b.type) {
+        const a_msg = message_store.get(a.last_msg_id);
+        const b_msg = message_store.get(b.last_msg_id);
+
+        if (a.type === "stream") {
+            return sort_comparator(a_msg.stream, b_msg.stream);
+        }
+        return sort_comparator(a_msg.display_reply_to, b_msg.display_reply_to);
+    }
+    // if type is not same sort between "private" and "stream"
+    return sort_comparator(a.type, b.type);
+}
+
 function topic_sort(a, b) {
-    const a_topic = message_store.get(a.last_msg_id).topic;
-    const b_topic = message_store.get(b.last_msg_id).topic;
-    if (a_topic > b_topic) {
-        return 1;
-    } else if (a_topic === b_topic) {
-        return 0;
+    const a_msg = message_store.get(a.last_msg_id);
+    const b_msg = message_store.get(b.last_msg_id);
+
+    if (a_msg.type === b_msg.type) {
+        if (a_msg.type === "private") {
+            return sort_comparator(a_msg.display_reply_to, b_msg.display_reply_to);
+        }
+        return sort_comparator(a_msg.topic, b_msg.topic);
+    } else if (a_msg.type === "private") {
+        return sort_comparator(a_msg.display_reply_to, b_msg.topic);
     }
-    return -1;
+    return sort_comparator(a_msg.topic, b_msg.display_reply_to);
```

Comment:
OK; I'm not sure I understand the bug though.

Reply:
Testing, all that I've observed is that resolved topics are sorted alphabetically after everything, presumably because the unicode check character is larger than the alphabet characters. This seems fine?

Reply:
@timabbott When using `util.strcmp` for comparing the resolved topics appear at top of the list in sorted manner
![image](https://user-images.githubusercontent.com/51414879/195973937-fad47b68-04be-4690-9f55-ee9b84099c5b.png)
And then comes the unsorted topics (in sorted manner). Is this the expected behaviour?

Reply:
OK yeah the question was unclear. I think what you're saying is replacing the `toLowerCase` comparisons we have with `util.strcmp` would cause resolved topics to be sorted first, which I think I agree we probably don't want. So I think we can stick with what you have in this PR.